lib/cocoapods/installer/project_cache/project_cache_analyzer.rb

```diff
@@ -155,7 +155,7 @@ def compute_dirty_targets(targets)
             support_files_dir_exists = File.exist? target.support_files_dir
             xcodeproj_exists = case target
                                when PodTarget
-                                 File.exist? sandbox.pod_target_project_path(target.pod_name)
+                                 File.exist? sandbox.pod_target_project_path(target.project_name)
```

Comment:
@sebastianv1 I think this is correct but it needs a test. I might need your help to write it.